tests/disaster-recovery/mdr/test_app_failover_and_relocate.py

```diff
@@ -0,0 +1,226 @@
+        If fenced, unfence the cluster and reboot nodes
+        """
+
+        def finalizer():
+            if self.drcluster_name and get_fence_state(self.drcluster_name) == "Fenced":
+                enable_unfence(self.drcluster_name)
+                gracefully_reboot_ocp_nodes(self.namespace, self.drcluster_name)
+
+        request.addfinalizer(finalizer)
+
+    @pytest.mark.parametrize(
+        argnames=["primary_cluster_down"],
+        argvalues=[
+            pytest.param(
+                False,
+                marks=pytest.mark.polarion_id(polarion_id_primary_up),
+                id="primary_up",
+            ),
+            pytest.param(
+                True,
+                marks=pytest.mark.polarion_id(polarion_id_primary_down),
+                id="primary_down",
+            ),
+        ],
+    )
+    def test_application_failover_and_relocate(
+        self,
+        setup_acm_ui,
+        primary_cluster_down,
+        nodes_multicluster,
+        dr_workload,
+        node_restart_teardown,
+    ):
+        """
+        Tests to verify application failover and relocate between managed clusters
+        There are two test cases:
+            1) Failover and relocate of only application between managed clusters
+            2) Failover to secondary cluster when primary cluster is DOWN and Relocate
+                back to primary cluster once it recovers
+
+        This test is also compatible to be run from UI,
+        pass the yaml conf/ocsci/dr_ui.yaml to trigger it.
+        """
+
+        if config.RUN.get("mdr_failover_via_ui"):
+            ocs_version = version.get_semantic_ocs_version_from_config()
+            if ocs_version <= version.VERSION_4_12:
+                logger.error(
+                    "ODF/ACM version isn't supported for Failover/Relocate operation"
+                )
+                raise NotImplementedError
+
+        acm_obj = AcmAddClusters()
+        workload = dr_workload(num_of_subscription=1)[0]
+
+        # Create application on Primary managed cluster
+        set_current_primary_cluster_context(workload.workload_namespace)
+        primary_cluster_index = config.cur_index
+        node_objs = get_node_objs()
+        primary_cluster_name = get_current_primary_cluster_name(
+            namespace=workload.workload_namespace
+        )
+        self.drcluster_name = primary_cluster_name
+        self.namespace = workload.workload_namespace
+
+        # Stop primary cluster nodes
+        if primary_cluster_down:
+            logger.info("Stopping primary cluster nodes")
+            nodes_multicluster[primary_cluster_index].stop_nodes(node_objs)
+
+        # Fenced the primary managed cluster
+        enable_fence(drcluster_name=self.drcluster_name)
+
+        # Application Failover to Secondary managed cluster
+        secondary_cluster_name = get_current_secondary_cluster_name(
+            workload.workload_namespace
+        )
+        if config.RUN.get("mdr_failover_via_ui"):
```

Comment:
When failover is done via UI, UI validation is missing for the same using function `verify_failover_relocate_status_ui`

Reply:
Pls highlight where?

Reply:
It was added but there was typo in conditional, corrected to `mdr_failover_via_ui`